mobile/lib/modules/campaign/bloc/campaign_register/campaign_register.bloc.dart

```diff
@@ -1,20 +1,38 @@
+import 'dart:developer';
+
 import 'package:equatable/equatable.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:mobile/common/constants/handle_status.enum.dart';
+import 'package:mobile/data/repositories/campaign.repository.dart';
 
 part 'campaign_register.event.dart';
 part 'campaign_register.state.dart';
 
 class CampaignRegisterBloc
     extends Bloc<CampaignRegisterEvent, CampaignRegisterState> {
-  CampaignRegisterBloc() : super(const CampaignRegisterState()) {
-    on<CampaignRegisterSuccess>(_onRegisterSuccess);
+  final CampaignRepository _campaignRepository;
+  CampaignRegisterBloc({
+    required CampaignRepository campaignRepository,
+  })  : _campaignRepository = campaignRepository,
+        super(
+          const CampaignRegisterState(
+            status: HandleStatus.initial,
+          ),
+        ) {
+    on<CampaignRegisterSubmit>(_onRegisterSubmit);
   }
 
-  Future<void> _onRegisterSuccess(
-    CampaignRegisterSuccess event,
+  Future<void> _onRegisterSubmit(
+    CampaignRegisterSubmit event,
     Emitter<CampaignRegisterState> emit,
   ) async {
-    await Future.delayed(const Duration(seconds: 3));
-    emit(const CampaignRegisterState.success());
+    try {
+      emit(const CampaignRegisterState.loading());
+      await _campaignRepository.joinCampaign(event.campaignID);
+      emit(const CampaignRegisterState.success());
+    } catch (e) {
+      log(e.toString());
```

Comment:
dont need to use `log(e.toString())`, use `addError` instead of, it will log in bloc observable, better than use vanila `log`